lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,38 @@
+module RSpec
+  module Core
+    # @private
+    class NoProfiler
+      def example_groups
+        {}
+      end
+    end
+
+    # @private
+    class Profiler
+      Formatters.register self, :example_group_started, :example_group_finished, :example_started
```

Comment:
> It's not a formatter, it's just using the callback system.

Well, it's registered as a formatter, and I thought that was what your definition of a formatter was...given that I've previously named a class `ExceptionFormatter` and you asked me to rename since it wasn't registered as a formatter.

Regardless, why can't this callback handling be moved into the existing `ProfilerFormatter`?  Seems like it would have full access to the data tracked by the callbacks without having to complicate config and/or notifications with it.

You say:

> Further it can't be merged because the logic is used by 2 formatters, this is removing repeated logic.

I don't understand.  Which 2 formatters are using this?  It's brand new anyway...  I really don't understand why this can't be merged into the other one.